docker-compose-junit-jupiter/src/integrationTest/java/com/palantir/docker/compose/DockerComposeExtensionIntegrationTest.java

```diff
@@ -16,169 +16,37 @@
-    public void stop_can_be_run_on_stopped_container() {
-        forEachContainer(containerName -> {
-            try {
-                Container container = docker.containers().container(containerName);
-
-                container.stop();
-                assertThat(container.state()).isEqualTo(State.DOWN);
-
-                container.stop();
-            } catch (IOException | InterruptedException e) {
-                propagate(e);
-            }
-        });
-    }
-
-    @Test
-    public void start_can_be_run_on_running_container() {
-        forEachContainer(containerName -> {
-            try {
-                Container container = docker.containers().container(containerName);
-
-                container.start();
-                assertThat(container.state()).isEqualTo(State.HEALTHY);
-            } catch (IOException | InterruptedException e) {
-                propagate(e);
-            }
-        });
-    }
-
-    @Test
-    public void can_kill_and_start_containers() {
-        forEachContainer(containerName -> {
-            try {
-                Container container = docker.containers().container(containerName);
-
-                container.kill();
-                assertThat(container.state()).isEqualTo(State.DOWN);
-
-                container.start();
-                assertThat(container.state()).isEqualTo(State.HEALTHY);
-            } catch (IOException | InterruptedException e) {
-                propagate(e);
-            }
-        });
-    }
-
-    @Test
-    public void kill_can_be_run_on_killed_container() {
-        forEachContainer(containerName -> {
-            try {
-                Container container = docker.containers().container(containerName);
-
-                container.kill();
-                assertThat(container.state()).isEqualTo(State.DOWN);
-
-                container.kill();
-            } catch (IOException | InterruptedException e) {
-                propagate(e);
-            }
-        });
-    }
+    private static int port;
 
     @Test
-    @Disabled("This test will not run on Circle CI because it does not currently support docker-compose exec.")
-    public void exec_returns_output() throws Exception {
-        assertThat(docker.exec(options(), CONTAINERS.get(0), arguments("echo", "hello"))).isEqualTo("hello");
+    @Order(1)
+    public void an_external_port_exists() {
+        port = docker.containers().container("db").port(5432).getExternalPort();
+        assertThat(port).isNotZero();
     }
 
     @Test
-    public void run_returns_output() throws Exception {
-        String actual = docker.run(
-                DockerComposeRunOption.options("--entrypoint", "echo"),
-                CONTAINERS.get(0),
-                DockerComposeRunArgument.arguments("hello"));
-        assertThat(actual).isEqualTo("hello");
+    @Order(2)
+    public void container_stays_up_between_tests() {
+        assertThat(docker.containers().container("db").port(5432).getExternalPort()).isEqualTo(port);
```

Comment:
docker tries to reuse ports when they become free, so if you really wanted to be a correct test it probably makes more sense to use container id instead